trie/triedb/pathdb/history.go

```diff
@@ -619,7 +619,8 @@ func truncateFromTail(db ethdb.Batcher, freezer *rawdb.ResettableFreezer, ntail
 	}
 	// Ensure that the truncation target falls within the specified range.
 	if otail > ntail || ntail > ohead {
-		return 0, fmt.Errorf("out of range, tail: %d, head: %d, target: %d", otail, ohead, ntail)
+		log.Warn("truncate from tail out of range", "tail:", otail, "head:", ohead, "target:", ntail)
+		return 0, nil
```

Comment:
This interface can remain unchanged.

Reply:
Exactly.
Theoretically, adding a conditional check in the outer layer can prevent out-of-bounds issues. However, setting the logic for repeated deletions as a critical error and directly causing a panic would be too strict. Therefore, we have also increased the tolerance at the interface level. Ideally, it should not enter this logic. We have added a warning log to capture such cases to see if there are any other anomalies, while ensuring that the program does not panic.

Reply:
Exactly.
Theoretically, adding a conditional check in the outer layer can prevent out-of-bounds issues. However, setting the logic for repeated deletions as a critical error and directly causing a panic would be too strict. Therefore, we have also increased the tolerance at the interface level. Ideally, it should not enter this logic. We have added a warning log to capture such cases to see if there are any other anomalies, while ensuring that the program does not panic.

Reply:
Ok, got it～